primer/react
# feat(DataTable): add Table.Skeleton

Batch: https://github.com/github/primer/issues/1892

This is currently a visual an API implementation of `Table.Skeleton`. The behavior is not finalized when it comes to how to indicate loading content. In the meantime, the cells themselves contain "Loading" as text content. Most of the styles for the Table cell skeleton items are lifted from TreeView.

One caveat to this API, it is currently bound to `Column` as it seemed natural to toggle between the two with an existing set of `columns`. It may be useful in the future to permit less restrictive types (like `Array<string>`) for usage.

### Testing & Reviewing

- Verify that the "With Loading" example is available on the storybook and included in the docs.json for DataTable
- Verify that the styles look as expected for the Table Skeleton
- Verify that `Table.Skeleton` is defined and its props are exported
- Verify that a test exists for rendering a `Table.Skeleton`
  - Note: I omitted a test for `rows` since it's a visual only change and I did not want to test structural changes in enzyme
